File: InterviewBit/Arrays/Noble Integer.cpp
int Solution::solve(vector<int> &A) {
    int N = A.size();
    sort(A.begin(), A.end());
    
    for(int i=0; i<N; i++) {
        if(A[i] == N-1-i && A[i] != A[i+1] || i == N-1 && A[N-1] == 0) {
            return 1;
        }
    }
    return -1;
}

File: InterviewBit/Arrays/Noble Integer.rb
class Solution
    # @param a : array of integers
    # @return an integer
    def solve(a)
        a.sort!
        a.each_with_index { |num, i| return 1 if num == a.length-i-1 && a[i] != a[i+1] }
        -1
    end
end
